central/globaldb/postgres.go

```diff
@@ -104,6 +98,52 @@ func GetPostgres() *pgxpool.Pool {
 	return postgresDB
 }
 
+// GetPostgresConfig - gets the configuration used to connect to Postgres
+func GetPostgresConfig() (map[string]string, *pgxpool.Config, error) {
+	centralConfig := config.GetConfig()
+	password, err := os.ReadFile(dbPasswordFile)
+	if err != nil {
+		log.Fatalf("pgsql: could not load password file %q: %v", dbPasswordFile, err)
+		return nil, nil, err
+	}
+	source := fmt.Sprintf("%s password=%s", centralConfig.CentralDB.Source, password)
+
+	config, err := pgxpool.ParseConfig(source)
+	if err != nil {
+		log.Fatalf("Could not parse postgres config: %v", err)
+		return nil, nil, err
+	}
+
+	sourceMap, err := ParseSource(source)
+	if err != nil {
+		log.Fatalf("Could not parse postgres config: %v", err)
+		return nil, nil, err
+	}
+
+	return sourceMap, config, nil
+}
+
+// ParseSource - parses source string into a map for easier use
+func ParseSource(source string) (map[string]string, error) {
+	if source == "" {
+		return nil, errors.New("source string is empty")
+	}
+
+	sourceSlice := strings.Fields(source)
+	sourceMap := make(map[string]string)
+	for _, pair := range sourceSlice {
+		// Due to the possibility that the password could potentially have an = we
+		// need to ensure that we get the entire password
+		configSetting := strings.SplitN(pair, "=", 2)
```

Comment:
We've got a Split2 function in stringutils that can probably just alleviate some of these. Not sure we need to check for empty field values